.github/workflows/fast-tests.yml

```diff
@@ -0,0 +1,122 @@
+name: Fast Tests
```

Comment:
Could you define "fast"? For instance, "Evaluate fast tests that runs under 5 minutes" as fast can be quite subjective.

Reply:
What I really mean is that all tests in this file must satisfy the following two properties: 
1. We want them to run against every PR
2. They do not require AWS credentials (We only run the tests that require credentials on code that has been approved and merged to main)

"Fast" seemed a reasonable way of referring to them, but some possible other names are: 
1. Integ Tests (and we then rename the "Integ Tests" suite to "End-to-end Tests", since that's more of what that other suite is).
2. Lightweight Integ Tests (in contrast with the other "Full integ tests")

If we go with this file becoming "Integ Tests" and the other one becoming "End-to-end Tests", then we can trigger this one with every PR and merge, and trigger the other one with every merge to `main`, and remove the duplicated tests between the two.

Reply:
Either name is fine, can you add a comment explaining the rationale you just described so that the context is clear for others? (for instance saying that this is a list of tests that dont require cred and runs at all PRs)

Reply:
Done, updated in the next revision